### PR TITLE
Add sample project workflow

### LINKED ISSUE #139: "New Sample Project" feature
Once the user clicks on the "..." option in the corner, add a new option called "New Project from Sample Template ..."
Screen Shot 2023-03-14 at 7 01 26 PM

This option should open a new tab that will be somewhat similar to the "Java Overview" one. (Install Java on your VS Code so you can try it out):
Screen Shot 2023-03-14 at 7 05 19 PM

## Requirements
- The title should be "New Project from Sample Couchbase Template
- We will organize it by language: Java, .Net, NodeJS, Golang, Scala Python

## Projects

**Java**
- Create Spring Data CRUD project ... (should clone https://github.com/couchbase-examples/java-springdata-quickstart)
- Create Spring Boot CRUD project ... (should clone  https://github.com/couchbase-examples/java-springboot-quickstart)
- Create Key-Value Transaction Demo project ... ( should clone https://github.com/couchbase-examples/java-transactions-quickstart )

**NodeJS**
- Create Ottoman CRUD project ... (should clone https://github.com/couchbase-examples/ottomanjs-quickstart)
- Create NodeJS CRUD project ... ( should clone https://github.com/couchbase-examples/nodejs-quickstart)
- Create NextJS CRUD project ... (https://github.com/couchbase-examples/nextjs-quickstart)

**.NET**
- Create ASP.NET CRUD project ... (Should clone https://github.com/couchbase-examples/aspnet-quickstart)
- Create ASP.NET Minimum API CRUD project ... (https://github.com/couchbase-examples/aspnet-quickstart-minapi)

**Python**
- Create Flask CRUD project ... ( should clone https://github.com/couchbase-examples/python-quickstart)

**Golang**
- Create Golang CRUD project ... (Should clone https://github.com/couchbase-examples/golang-quickstart)

**Scala**
- Create Akka CRUD Project ... (Should clone https://github.com/couchbase-examples/scala-quickstart)

## New Sample Project Flow

Once the user clicks on one of the options above, we should ask for the name of the project (~similar to the image below)
Screen Shot 2023-03-14 at 8 15 04 PM

Then ask the location where the project should be saved:

Screen Shot 2023-03-14 at 8 15 17 PM

Then we clone the project and ask the user if he wants to open it:
Screen Shot 2023-03-14 at 8 13 36 PM

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Describe the problem this PR is solving
Adds a sample project workflow to allow extension uses to clone a Couchbase sample project and open / add it to their workspace. The extension asks for a project name and local folder to put the example then downloads the example. Once the example has been downloaded, the extension asks if they would like to open the example or add to the current workspace.

- Closes #139 

## Short description of the changes
- Adds gitly as extension dependency
- Add and wire up new action in package.json & extension.js
- Add webview that shows a list of sample projects grouped by language
- Add workflow to the click event of selecting one of the examples

TODO:
- [x] Improve styling of sample webview

### CODE REVIEW

**Comment by AayushTyagi1** [comment]
@MikeGoldsmith
Bug Report: Uncaught Error "An instance of the VS Code API has already been acquired" when Clicking on Project Links after Cancelling while it ask for Project Name

To reproduce this issue, please follow these steps:

- Click on "New Project from Sample Template..." in the interface.
- Select one of the available links.
- When prompted to enter a project name, cancel the operation by pressing the Esc key.
- Observe the warning message "Project name is required," which is the expected behaviour.
- Attempt to click on any other project within the same window.
- At this point, you will encounter the "Uncaught Error: An instance of the VS Code API has already been acquired" error.

Please investigate this issue and take appropriate action to resolve it. Let me know if you require any additional information or steps to reproduce the bug.

File: src/extension.ts [hunk 1]
```diff
@@ -863,6 +865,84 @@ export function activate(context: vscode.ExtensionContext) {
...
+                  { uri: projectUri }
+                );
+                break;
+            }
+          },
+          undefined,
+          context.subscriptions
+        );
+      } catch (err) {
+        logger.error("Failed to open Query Workbench");
```
Comment (AayushTyagi1) [inline src/extension.ts:940]:
```suggestion
        logger.error("Failed to open sample project template");
```

File: src/extension.ts [hunk 2]
```diff
@@ -863,6 +865,84 @@ export function activate(context: vscode.ExtensionContext) {
...
+                );
+                break;
+              case "Add to Workspace":
+                vscode.workspace.updateWorkspaceFolders(
+                  vscode.workspace.workspaceFolders
+                    ? vscode.workspace.workspaceFolders.length
+                    : 0,
+                  null,
+                  { uri: projectUri }
+                );
```
Comment (AayushTyagi1) [inline src/extension.ts:932]: Are we suppose to close the sampleProject panel after adding project to the workspace
If yes please close it
If no, then here also we are facing the same error as https://github.com/couchbaselabs/VSCode-Couchbase/pull/208#issuecomment-1549263921

Comment (MikeGoldsmith) [reply src/extension.ts:932]: Thanks @AayushTyagi1 - added a commit to close the panel after a sample has been crated in [1b70e9e](https://github.com/couchbaselabs/VSCode-Couchbase/pull/208/commits/1b70e9e37fe131bdcd16d251cad456a02414924c)

**Comment by AayushTyagi1** [comment]
Abnormal behaviour if I choose already user folder name for my new project
Here's a step-by-step description of the issue:

- When creating a new project, let's say "test1" with Python as the chosen language.
- Subsequently, attempting to create another project in the same folder, this time with Java as the language and again using the name "test1".
- The result is that the files from the new Java project overlap with the existing Python project in the project structure, leading to an unintended combination of both Python and Java files within the same folder.
- To provide a more desirable user experience, it would be beneficial if an error message could be displayed, notifying the user that a project with the name "test1" already exists in the selected folder. This would prompt the user to either choose a different name for the project or select an alternative folder to avoid file overlap.

**Comment by MikeGoldsmith** [comment]
@AayushTyagi1 I've updated the workflow to check if the project uri already exists, and show an error it does. This prevents a user from re-using an existing directory for the sample to be cloned into.

**Review by AayushTyagi1 — APPROVED** [review]

**Pull request merged**